core/capabilities/remote/trigger_publisher.go

```diff
@@ -70,34 +56,33 @@ func NewTriggerPublisher(config *commoncap.RemoteTriggerConfig, underlying commo
 		config = &commoncap.RemoteTriggerConfig{}
 	}
 	config.ApplyDefaults()
-	membersCache := make(map[uint32]map[p2ptypes.PeerID]bool)
-	for id, don := range workflowDONs {
-		cache := make(map[p2ptypes.PeerID]bool)
-		for _, member := range don.Members {
-			cache[member] = true
-		}
-		membersCache[id] = cache
-	}
-	return &triggerPublisher{
+
+	publisher := &triggerPublisher{
 		config:          config,
 		underlying:      underlying,
 		capInfo:         capInfo,
 		capDonInfo:      capDonInfo,
 		workflowDONs:    workflowDONs,
-		membersCache:    membersCache,
 		dispatcher:      dispatcher,
-		messageCache:    messagecache.NewMessageCache[registrationKey, p2ptypes.PeerID](),
-		registrations:   make(map[registrationKey]*pubRegState),
 		batchingQueue:   make(map[[32]byte]*batchedResponse),
 		batchingEnabled: config.MaxBatchSize > 1 && config.BatchCollectionPeriod >= minAllowedBatchCollectionPeriod,
 		stopCh:          make(services.StopChan),
 		lggr:            lggr.Named("TriggerPublisher"),
 	}
+
+	registrationServer := registration.NewServer(lggr, publisher, capInfo, config.RegistrationExpiry, workflowDONs, "TriggerPublisher")
+
+	publisher.registrationServer = registrationServer
```

Comment:
just put those two lines above object creation and return it like it was before - why make an exception for this one field?